content/news/posts/2025/breaking-change-guzzle-removal-v9.md

```diff
@@ -0,0 +1,77 @@
+
+Since PrestaShop no longer relies on Guzzle, keeping it in the core is unnecessary. Additionally, dependency management has been a recurring issue. In PrestaShop 8, upgrading Guzzle from version 5 to 7 already caused compatibility problems for modules. Some developers had to bundle their own Guzzle version, but when multiple modules included different versions, it led to conflicts. To prevent such issues and improve overall stability, we decided to remove Guzzle entirely.
+
+## How can I adapt my module?
+
+With Guzzle no longer available by default, you must adjust your module. Here are a few solutions to replace it effectively.
+
+### Use Symfony HTTP Client (recommended)
+
+The best way to avoid conflicts is to minimize external dependencies and rely on core libraries whenever possible. Since Symfony is the foundation of PrestaShop and will continue to be so in the future, it's a safe and stable dependency that won't be removed.
+
+Symfony includes an HTTP client component, which has been integrated into PrestaShop since version 1.7.7. If your module targets PrestaShop 1.7.7 or later, you can safely use the Symfony HTTP Client. You can find more details on how to implement it in the [Symfony documentation](https://symfony.com/doc/current/http_client.html).
+
+```php
+use Symfony\Component\HttpClient\HttpClient;
+
+$client = HttpClient::create();
+$response = $client->request('GET', 'https://api.domain.com');
+```
+
+### Use a custom HTTP client implementation
+
+You can create your own HTTP client, ensuring complete control and eliminating the risk of conflicts with other modules using different library versions.
+
+However, the downside is that you'll need to maintain it yourself, which can lead to unnecessary complexity and duplicating existing solutions.
+
+Here's an example of a custom HTTP client used in the PrestaShop EventBus module: [PrestaShop EventBus HTTP Client](https://github.com/PrestaShopCorp/ps_eventbus/blob/27ca0e360d88345e57e945ad6369a215f4e615d0/src/Api/HttpClient.php)
+
+### Forcefully include Guzzle in my module's vendors (not recommended)
+
+This is the **quick and dirty solution** - it allows you to keep your existing code without modifications. However, as mentioned earlier, we strongly discourage this approach because it increases the risk of conflicts with other modules that may include different Guzzle versions.
+
+If you're developing a custom module for a specific shop with a controlled environment and a limited set of modules, this might work as a short-term fix. But if you distribute your module (e.g., on the PrestaShop Marketplace) for multiple shops, avoid this solution, as it can lead to unpredictable compatibility issues.
+
+## What if I want my module to work across multiple PrestaShop versions?
+
+Since the Symfony HTTP Client is only available from PrestaShop 1.7.7 onward, you'll need to adapt your code based on the PrestaShop version your module is running on.
```

Comment:
```suggestion
Since the Symfony HTTP client is only available from PrestaShop 1.7.7 onward, you'll need to adapt your code based on the PrestaShop version your module is running on.
```

Reply:
HTTP Client is a package name, so in my opinion we should respect the name "HTTP Client" @coffrep - what do you think?

Reply:
Hi @kpodemski I went on the [Symfony site](https://symfony.com/doc/current/http_client.html) to see how they write "HTTP client" and it seems as though they write it with a lower case "c"? But if you feel as though it might need a capital "C", I'd be fine with that.

Reply:
ahh, right, I think the capital letter is only used in the title of the page :) in that case we can keep it consistent and use the lower case "c"

Reply:
Great, thanks a lot @kpodemski for the comment!